cli/lsp/config.rs

```diff
@@ -1083,512 +1094,468 @@ impl LspTsConfig {
 }
 
 impl ConfigData {
   async fn load(
-    config_file_specifier: Option<&ModuleSpecifier>,
+    specified_config: Option<&Path>,
     scope: &ModuleSpecifier,
-    workspace_root: Option<&ConfigData>,
     settings: &Settings,
-    file_fetcher: Option<&Arc<FileFetcher>>,
+    file_fetcher: &Arc<FileFetcher>,
+    // sync requirement is because the lsp requires sync
+    cached_deno_config_fs: &(dyn DenoConfigFs + Sync),
+    deno_json_cache: &(dyn DenoJsonCache + Sync),
+    pkg_json_cache: &(dyn PackageJsonCache + Sync),
   ) -> Self {
-    if let Some(specifier) = config_file_specifier {
-      match ConfigFile::from_specifier(
-        &DenoConfigFsAdapter::new(&RealFs),
-        specifier.clone(),
-        &deno_config::ConfigParseOptions::default(),
-      ) {
-        Ok(config_file) => {
-          lsp_log!(
-            "  Resolved Deno configuration file: \"{}\"",
-            config_file.specifier.as_str()
-          );
-          Self::load_inner(
-            Some(config_file),
-            scope,
-            workspace_root,
-            settings,
-            file_fetcher,
-          )
-          .await
-        }
-        Err(err) => {
-          lsp_warn!(
-            "  Couldn't read Deno configuration file \"{}\": {}",
-            specifier.as_str(),
-            err
-          );
-          let mut data = Self::load_inner(
-            None,
-            scope,
-            workspace_root,
-            settings,
-            file_fetcher,
-          )
-          .await;
-          data
-            .watched_files
-            .insert(specifier.clone(), ConfigWatchedFileType::DenoJson);
-          let canonicalized_specifier = specifier
-            .to_file_path()
-            .ok()
-            .and_then(|p| canonicalize_path_maybe_not_exists(&p).ok())
-            .and_then(|p| ModuleSpecifier::from_file_path(p).ok());
-          if let Some(specifier) = canonicalized_specifier {
-            data
-              .watched_files
-              .insert(specifier, ConfigWatchedFileType::DenoJson);
+    let scope = Arc::new(scope.clone());
+    let discover_result = match scope.to_file_path() {
+      Ok(scope_dir_path) => {
+        let paths = [scope_dir_path];
+        Workspace::discover(
+          match specified_config {
+            Some(config_path) => {
+              deno_config::workspace::WorkspaceDiscoverStart::ConfigFile(
+                config_path,
+              )
+            }
+            None => {
+              deno_config::workspace::WorkspaceDiscoverStart::Paths(&paths)
+            }
+          },
+          &WorkspaceDiscoverOptions {
+            fs: cached_deno_config_fs,
+            additional_config_file_names: &[],
+            deno_json_cache: Some(deno_json_cache),
+            pkg_json_cache: Some(pkg_json_cache),
+            discover_pkg_json: !has_flag_env_var("DENO_NO_PACKAGE_JSON"),
+            config_parse_options: Default::default(),
+            // todo: provide this value
```

Comment:
Do you want to address this before landing?

Reply:
Oh yeah, I forgot. Thanks!

Reply:
Oh wait, this is derived solely from the config. I'll remove this todo.